public/app/features/dashboard/components/Inspector/PanelInspector.tsx

```diff
@@ -1,42 +1,187 @@
+}
+
+export class PanelInspector extends PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      data: [],
+      selected: 0,
+      tab: InspectTab.Data,
+    };
+  }
+
+  async componentDidMount() {
+    const { panel } = this.props;
+    if (!panel) {
+      this.onDismiss(); // Try to close the component
+      return;
+    }
+
+    // TODO? should we get the result with an observable once?
+    const lastResult = (panel.getQueryRunner() as any).lastResult;
+    if (!lastResult) {
+      this.onDismiss(); // Usually opened from refresh?
+      return;
+    }
+
+    // Find the first DataSource wanting to show custom metadata
+    let metaDS: DataSourceApi;
+    const data = lastResult?.series as DataFrame[];
+    if (data) {
+      for (const frame of data) {
+        const key = frame.meta?.ds?.dataSourceName;
+        if (key) {
+          const ds = await getDataSourceSrv().get(key);
+          if (ds && ds.components.MetadataInspector) {
+            metaDS = ds;
+            break;
+          }
+        }
+      }
+    }
+
+    // Set last result, but no metadata inspector
+    this.setState({
+      last: lastResult,
+      data,
+      metaDS,
+    });
+  }
+
   onDismiss = () => {
     getLocationSrv().update({
       query: { inspect: null },
       partial: true,
     });
   };
 
+  onSelectTab = (item: SelectableValue<InspectTab>) => {
+    this.setState({ tab: item.value || InspectTab.Data });
+  };
+
+  onSelectedFrameChanged = (item: SelectableValue<number>) => {
+    this.setState({ selected: item.value || 0 });
+  };
+
+  renderMetaDataInspector() {
+    const { metaDS, data } = this.state;
+    if (!metaDS || !metaDS.components?.MetadataInspector) {
+      return <div>No Metadata Inspector</div>;
+    }
+    return <metaDS.components.MetadataInspector datasource={metaDS} data={data} />;
+  }
+
+  renderDataTab() {
+    const { data, selected } = this.state;
+    if (!data || !data.length) {
+      return <div>No Data</div>;
+    }
+    const choices = data.map((frame, index) => {
+      return {
+        value: index,
+        label: `${frame.name} (${index})`,
+      };
+    });
+
+    // Apply dummy styles
+    const processed = applyFieldOverrides({
+      data,
+      theme: config.theme,
+      fieldOptions: { defaults: {}, overrides: [] },
+      replaceVariables: (value: string) => {
+        return value;
+      },
+    });
+
+    return (
+      <div>
+        {choices.length > 1 && (
+          <div>
+            <Select
+              options={choices}
+              value={choices.find(t => t.value === selected)}
+              onChange={this.onSelectedFrameChanged}
+            />
+          </div>
+        )}
+        <div style={{ border: '1px solid #666' }}>
+          <Table width={330} height={400} data={processed[selected]} />
+        </div>
+      </div>
+    );
+  }
+
+  renderIssueTab() {
+    return <div>TODO: show issue form</div>;
```

Comment:
Is this for reporting issues to github?

Reply:
Yes -- I would love some way to submit an issue that includes all the stuff we want:

* panel json
* query result (anonymized?)
* installed plugins
* etc

it could either format text that could be pasted into github, or submit a payload to an API on grafana.com